src/main/java/rx/observables/SyncOnSubscribe.java

```diff
@@ -0,0 +1,499 @@
+        return new LambdaOnSubscribe<S, T>(generator, next, onUnsubscribe);
+    }
+
+    /**
+     * Generates a synchronous onSubscribe function that manages back pressure requests delegating
+     * to your {@code next} function/closure to produce data when values are requested by the
+     * downstream subscriber.
+     * 
+     * This overload creates a "state-less" SyncOnSubscribe which does not have an explicit state
+     * value.
+     * 
+     * @param next
+     *            produces data to the downstream subscriber (see {@link #next(Object, Subscriber)
+     *            next(S, Subscriber)})
+     * @return an OnSubscribe that emits data downstream in a protocol compatible with
+     *         back-pressure.
+     */
+    @Experimental
+    public static <T> SyncOnSubscribe<Void, T> createStateless(final Action1<Subscriber<? super T>> next) {
+        Func2<Void, Subscriber<? super T>, Void> nextFunc = new Func2<Void, Subscriber<? super T>, Void>() {
+            @Override
+            public Void call(Void state, Subscriber<? super T> subscriber) {
+                next.call(subscriber);
+                return state;
+            }
+        };
+        return new LambdaOnSubscribe<Void, T>(VOID_GENERATOR, nextFunc, VOID_ON_UNSUBSCRIBE);
+    }
+
+    /**
+     * Generates a synchronous onSubscribe function that manages back pressure requests delegating
+     * to your {@code next} function/closure to produce data when values are requested by the
+     * downstream subscriber.
+     * 
+     * This overload creates a SyncOnSubscribe without an explicit state.
+     * 
+     * @param next
+     *            produces data to the downstream subscriber (see {@link #next(Object, Subscriber)
+     *            next(S, Subscriber)})
+     * @return an OnSubscribe that emits data downstream in a protocol compatible with
+     *         back-pressure.
+     */
+    @Experimental
+    public static <T> SyncOnSubscribe<Void, T> createStateless(final Action1<Subscriber<? super T>> next, final Action0 onUnsubscribe) {
+        Func0<Void> generator = new Func0<Void>() {
+            @Override
+            public Void call() {
+                return null;
+            }
+        };
+        Func2<Void, Subscriber<? super T>, Void> nextFunc = new Func2<Void, Subscriber<? super T>, Void>() {
+            @Override
+            public Void call(Void state, Subscriber<? super T> subscriber) {
+                next.call(subscriber);
+                return null;
+            }
+        };
+        Action1<Void> wrappedOnUnsubscribe = new Action1<Void>() {
+            @Override
+            public void call(Void t) {
+                onUnsubscribe.call();
+            }
+        };
+        return new LambdaOnSubscribe<Void, T>(generator, nextFunc, wrappedOnUnsubscribe);
+    }
+
+    private static final Action1<Void> VOID_ON_UNSUBSCRIBE = new Action1<Void>() {
```

Comment:
`null` onUnsubscribe is allowed so no need for this (and would cause optimal path not taken).